stor/dx.py

```diff
@@ -176,22 +178,59 @@ def joinpath(self, *others):
         """
         return self.path_class(self.path_module.join(self.normpath(), *others))
 
+    def to_url(self):
+        """For compatibility with OBS - returns ``temp_url()``"""
+        return self.temp_url()
+
     def temp_url(self, lifetime=300, filename=None):
         """Obtains a temporary URL to a DNAnexus data-object.
 
+        If ``DX_FILE_PROXY_URL`` or ``[dx] file_proxy_url=`` is set, will use that to construct a
+        path instead, e.g.::
+
+            >>> stor.Path('dx://proj:/folder/mypath.csv').temp_url()
+            'https://dl.dnanex.us/F/D/awe1323/mypath.csv'
+            >>> with stor.settings.use({'dx': {'file_proxy_url':
+            ...     'https://myproxy.example.com/gateway'}):
+            ... stor.Path('dx://proj:/folder/mypath.csv').temp_url()
+            'https://my-dnax-proxy.example.com/gateway/proj/folder/mypath.csv'
+
+        The file proxy is assumed to be a service that, when given DX path and project, will proxy
+        through to DNAnexus to render content.
+
         Args:
             lifetime (int): The time (in seconds) the temporary
-                URL will be valid
+                URL will be valid (only for temp URL generation)
             filename (str, optional): A urlencoded filename to use for
                 attachment, otherwise defaults to object name
+                (to bypass, set to empty string).
```

Comment:
```suggestion
                (to bypass, set to empty string). Has to be the same as actual object name if `file_proxy_url` is set.
```

Reply:
do you think important to repeat it both here and under `Raises` section?

Reply:
I think it's useful, b/c you don't always read raises section. Up to you though.